crates/shared/src/trade_finding/oneinch.rs

```diff
@@ -24,6 +28,8 @@ struct Inner {
     disabled_protocols: Vec<String>,
     protocol_cache: ProtocolCache,
     referrer_address: Option<H160>,
+    spender_cache: Arc<Mutex<(H160, Instant)>>,
```

Comment:
Instead of making this `Arc` you can `Arc<Inner>`. No strong feelings though.

Reply:
Addressed in [e989660](https://github.com/cowprotocol/services/pull/745/commits/e989660213e7dd8b858443dc25dce71f9d871751).